edx_when/field_data.py

```diff
@@ -55,9 +56,10 @@ def _load_dates(self, course_id, user, use_cached=True):
         """
         Load the dates from the database.
         """
-        dates = {}
-        for (location, field), date in api.get_dates_for_course(course_id, user, use_cached=use_cached).items():
-            dates[text_type(location), field] = date
+        with read_queries_only():
```

Comment:
This needs a corresponding edx-platform change to use the new router before it works, right?

Reply:
Yes